packages/statemanager/src/statefulVerkleStateManager.ts

```diff
@@ -0,0 +1,467 @@
+    this._debug = debugDefault('statemanager:statefulVerkleStatemanager')
+    this.originalStorageCache = new OriginalStorageCache(this.getStorage.bind(this))
+    this._caches = opts.caches
+    this.keccakFunction = opts.common?.customCrypto.keccak256 ?? keccak256
+    this.verkleCrypto = opts.verkleCrypto
+  }
+
+  getAccount = async (address: Address): Promise<Account | undefined> => {
+    const elem = this._caches?.account?.get(address)
+    if (elem !== undefined) {
+      return elem.accountRLP !== undefined
+        ? createPartialAccountFromRLP(elem.accountRLP)
+        : undefined
+    }
+
+    const stem = getVerkleStem(this.verkleCrypto, address, 0)
+
+    // First retrieve the account "header" values from the trie
+    const accountValues = await this._trie.get(stem, [
+      VerkleLeafType.BasicData,
+      VerkleLeafType.CodeHash,
+    ])
+
+    let account
+    if (accountValues[0] !== undefined) {
+      const basicData = decodeVerkleLeafBasicData(accountValues[0]!)
+      account = createPartialAccount({
+        version: basicData.version,
+        balance: basicData.balance,
+        nonce: basicData.nonce,
+        codeHash:
+          accountValues[1] === undefined
+            ? KECCAK256_NULL // codeHash is undefined in trie (i.e. not touched)
+            : equalsBytes(accountValues[1], new Uint8Array(32))
+              ? KECCAK256_NULL // codeHash is deleted in trie (i.e. overwritten with zeroes)
+              : accountValues[1],
+        codeSize: basicData.codeSize,
+        storageRoot: KECCAK256_NULL, // TODO: Add storage stuff
+      })
+    }
+    // check if the account didn't exist if any of the basic keys are undefined
+    else if (accountValues[0] === undefined || accountValues[1] === undefined) {
+      if (this.DEBUG) {
+        this._debug(`getAccount address=${address.toString()} from DB (non-existent)`)
+      }
+      this._caches?.account?.put(address, account)
+    }
+
+    if (this.DEBUG) {
+      this._debug(`getAccount address=${address.toString()} stem=${short(stem)}`)
+    }
+    return account
+  }
+
+  putAccount = async (address: Address, account?: Account): Promise<void> => {
+    if (this.DEBUG) {
+      this._debug(
+        `Save account address=${address} nonce=${account?.nonce} balance=${
+          account?.balance
+        } contract=${account && account.isContract() ? 'yes' : 'no'} empty=${
+          account && account.isEmpty() ? 'yes' : 'no'
+        }`,
+      )
+    }
+    if (this._caches?.account === undefined) {
+      if (account !== undefined) {
+        const stem = getVerkleStem(this.verkleCrypto, address, 0)
+        const basicDataBytes = encodeVerkleLeafBasicData(account)
+        await this._trie.put(
+          stem,
+          [VerkleLeafType.BasicData, VerkleLeafType.CodeHash],
+          [basicDataBytes, account.codeHash],
+        )
+      } else {
+        // Delete account
```

Comment:
Is this a TODO?

Reply:
AH, yes, good catch.  Will fix this tomorrow